### PR TITLE
[development] Promremote storage local setup

### PR DESCRIPTION
**Pull request by Antanukas**
**What this PR does / why we need it**: Local dev setup to quickly test Prom Remote Write storage.

**Special notes for your reviewer**:

Refer to readme that describes this setup.

In order to generate some realistic metrics it's possible to scrape CAdvisor metrics from Kubernetes.
This is achieved by using `kubectl proxy` and the fact that each node exposes its metrics at `localhost:8001/api/v1/nodes/${node}/proxy/metrics/cadvisor`.

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
None
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
None
```

### CODE REVIEW

File: scripts/development/m3_prom_remote_stack/start_m3.sh [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    docker-compose -f docker-compose.yml up --build $DOCKER_ARGS m3coordinator01
+else
+    docker-compose -f docker-compose.yml up $DOCKER_ARGS m3coordinator01
+fi
+
+echo "Wait for coordinator API to be up"
+ATTEMPTS=10 MAX_TIMEOUT=4 TIMEOUT=1 retry_with_backoff  \
+  'curl -vvvsSf localhost:7201/health'
+
+    echo "Running aggregator pipeline"
```
Comment (vpranckaitis) [inline scripts/development/m3_prom_remote_stack/start_m3.sh:38]: nit:
```suggestion
echo "Running aggregator pipeline"
```

File: scripts/development/m3_prom_remote_stack/start_m3.sh [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+prepare_build_cmd() {
+    build_cmd="cd $RELATIVE && make clean-build docker-dev-prep && cp -r ./docker ./bin/ && $1"
+}
+DOCKER_ARGS="-d --renew-anon-volumes"
+
+M3COORDINATOR_DEV_IMG=$(docker images m3coordinator:dev | fgrep -iv repository | wc -l | xargs)
+M3AGGREGATOR_DEV_IMG=$(docker images m3aggregator:dev | fgrep -iv repository | wc -l | xargs)
+M3COLLECTOR_DEV_IMG=$(docker images m3collector:dev | fgrep -iv repository | wc -l | xargs)
+
+    docker-compose -f docker-compose.yml up $DOCKER_ARGS etcd01
```
Comment (vpranckaitis) [inline scripts/development/m3_prom_remote_stack/start_m3.sh:20]: nit:
```suggestion
docker-compose -f docker-compose.yml up $DOCKER_ARGS etcd01
```

File: scripts/development/m3_prom_remote_stack/start_m3.sh [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+curl -vvvsSf -X POST -H "Topic-Name: aggregator_ingest" -H "Cluster-Environment-Name: default_env" localhost:7201/api/v1/topic -d '{
+"consumerService": {
+    "serviceId": {
+    "name": "m3aggregator",
+    "environment": "default_env",
+    "zone": "embedded"
+    },
+    "consumptionType": "REPLICATED",
+    "messageTtlNanos": "600000000000"
+}
```
Comment (vpranckaitis) [inline scripts/development/m3_prom_remote_stack/start_m3.sh:71]: nit:
```suggestion
    "consumerService": {
        "serviceId": {
        "name": "m3aggregator",
        "environment": "default_env",
        "zone": "embedded"
        },
        "consumptionType": "REPLICATED",
        "messageTtlNanos": "600000000000"
    }
```

File: scripts/development/m3_prom_remote_stack/start_m3.sh [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+echo "Validating M3Coordinator topology"
+[ "$(curl -sSf localhost:7201/api/v1/services/m3coordinator/placement | jq .placement.instances.m3coordinator01.id)" == '"m3coordinator01"' ]
+echo "Done validating topology"
+
+# Do this after placement for m3coordinator is created.
+echo "Adding coordinator as a consumer to the aggregator outbound topic"
+curl -vvvsSf -X POST -H "Topic-Name: aggregated_metrics" -H "Cluster-Environment-Name: default_env" localhost:7201/api/v1/topic -d '{
+    "consumerService": {
+            "serviceId": {
+            "name": "m3coordinator",
```
Comment (vpranckaitis) [inline scripts/development/m3_prom_remote_stack/start_m3.sh:113]:
```suggestion
    "consumerService": {
        "serviceId": {
            "name": "m3coordinator",
```

**Review by vpranckaitis — APPROVED** [review]
LGTM, just a few nits regarding indentation. Though I don't know all the intricacies of running M3 locally, and this applies even more for Prometheus.

**Review by robskillington — APPROVED** [review]
LGTM

**Pull request merged**